README.md

```diff
@@ -1,2 +1,307 @@
+
+```bash
+qiime diversity core-metrics-phylogenetic \
+  --i-phylogeny tree.qza \
+  --i-table table.qza \
+  --p-sampling-depth 1000 \
+  --m-metadata-file metadata.tsv \
+  --output-dir .
+```
+
+### Alignment ambiguity
+
+It is quite common that one query sequence can be aligned to multiple reference genomes. In such cases, Woltka by default counts each gOTU as 1 / _k_, where _k_ is the total number of matching genomes.
+
+Alternatively, one may choose to discard all non-unique matches, by adding a flag:
+
+```bash
+woltka gotu --no-ambig ...
+```
+
+### Custom alignment
+
+Technically, one can use any sequence aligners and reference genome databases to generate alignment files which can then be converted into a gOTU table. We cannot validate the goodness of outcome, but understand that you may have this intention considering the consistency with existing parts of your analytical pipeline. For examples:
+
+```bash
+bwa mem refseq.fna input.R1.fq input.R2.fq > output.sam
+```
+
+```bash
+blastn -db refseq_genomes -query input.fa -max_target_seqs 1 -outfmt 6 -out output.txt
+```
+
+However, most of these protocols generate mappings of reads to nucleotides (e.g., chromosomes or scaffolds), rather than to genomes. In order to produce gOTUs, one needs to supply Woltka with a nucleotide-to-genome mapping file:
+
+```bash
+woltka gotu --map nucl2g.txt ...
+```
+
+
+## Input files
+
+The input files for Woltka are sequence **alignment** files. The term "alignment" here describes the operation of aligning short sequencing reads against long reference sequences. The basic information an alignment file provides is the **mapping** between queries and subjects. In addition, the position and quality of alignments are available in some formats, which are useful in some applications.
+
+### File formats
+
+Woltka supports the following alignment formats (specified by parameter `--format` or `-f`):
+
+- `map`: Simple mapping of query \<tab\> subject.
+- `sam`: SAM format. Supported by multiple tools such as Bowtie2 and BWA.
+- `b6o`: BLAST tabular format (i.e., BLAST parameter `-outfmt 6`). Supported by multiple tools such as BLAST, DIAMOND, VSEARCH, BURST, etc.
+
+If not specified, Woltka will automatically infer the format of input alignment files.
+
+Woltka supports and automatically detects common file compression formats including `gzip`, `bzip2` and `xz`.
+
+### Filename patterns
+
+In the default mode, Woltka treats every file under the directory specified by the `--input` or `-i` parameter as an alignment file for one sample. The sample ID is automatically extracted from the filename, with filename extension stripped. Compression file extensions are automatically recognized. For example, the sample IDs for `S01.sam` and `S02.m8.gz` are `S01` and `S02`, respectively.
```

Comment:
How common is this naming convention?

Unless it is _exceptionally_ common, I think it could be better to make the user be more explicit, e.g., providing a file with `sample_id    alignment_file` pairs. I think in instances where a lot of compute is necessary, inferring the wrong convention could be particularly bad.

Reply:
This is a good idea. I can add a feature to allow for `sample_id <tab> alignment_file` mapping files. (Note: this feature only applies to non-demultiplexed alignments.)

***

The convention of dealing with multiple files with stem filenames as sample IDs is *somehow* common. For example in MetaPhlAn2:

```
merge_metaphlan_tables.py *.txt > table.tsv
```

In Kraken:

```
kraken-mpa-report --header-line --db $db sample1.output sample2.output ... > kraken.mpa.reports
```

***

A design question is: should the program be "smart" or "strict"? I totally understand the proposal of making things more explicit, and I also had this consideration when designing the program. Eventually I thought that making usage simple is important. A mandatory configuration file (here a sample-to-alignment mapping file) may be quite blocking to beginners.

Back to the Oecophylla time, the most challenging step for beginners was to generate a mapping of sample IDs to FastQ files. That was more challenging than the current scenario because it is in YAML format and one sample may correspond to multiple FastQ files (i.e., multiple lanes) in both forward and reverse files. Oecophylla implements a "smart" function which directly reads a sample sheet to get these sample-FastQ pairs. However it won't work if the user wants to combine runs or something.

So I think having everything done via command-line parameters may be favorable. Meanwhile it is also important, as you suggested, to have a more controlled behavior as long as the user wants.

***

I think a workaround which balances both strategies is to let the program print out all automatically determined parameter status when executing.